odl/operator/oputils.py

```diff
@@ -319,6 +319,76 @@ def rmatvec(v):
                                               dtype=dtype)
 
 
+def as_scipy_functional(func, return_gradient=False):
+    """Wrap ``op`` as a function operating on linear arrays.
+
+    This is intended to be used with the scipy solvers.
+
+    Parameters
+    ----------
+    func : `Functional`
+        A functional that should be wrapped
+    return_gradient : bool, optional
+        If the gradient of the functional should also be returned
+
+    Returns
+    -------
+    function : ``callable``
+        The wrapped functional.
+    gradient : ``callable``, optional
```

Comment:
Yeah, but the user does not have this option, at least not in principle. Here it happens to be that case. At least potentially it's outside the control of the user, so it's rather a "maybe" or "may not be returned". 
But let's not be too orthodox. If you happen to come across a better formulation, take that, otherwise just go ahead.